apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.InsertData.cs

```diff
@@ -144,13 +144,17 @@ public override void InsertRows(TableReference tableReference, IEnumerable<BigQu
                 return;
             }
             var response = request.Execute();
-            HandleInsertAllResponse(response);
+            HandleInsertAllResponse(response, options);
         }
 
-        private void HandleInsertAllResponse(TableDataInsertAllResponse response)
+        private void HandleInsertAllResponse(TableDataInsertAllResponse response, InsertOptions options)
         {
             var errors = response.InsertErrors;
-            if (errors?.Count > 0)
+            bool shouldThrow = options == null ||
```

Comment:
I would simplify this - we should throw if we haven't been told not to. If we keep the options entirely independent, it makes it easier to reason about for both us and callers. That'll be particularly important if there are other skip options later.

Reply:
Yes, I thought about making them independent as well. The reason I didn't was because then there might be cases in which not even good rows are inserted but nothing is thrown either, albeit at the request of the caller, and allowing that case seemed somewhat odd.
But agree that making them independent will simplify things a lot. I'll make a point of explaining the Supress... property better.